cpp/include/cudf/lists/gather.hpp

```diff
@@ -43,7 +43,7 @@ namespace lists {
  * @endcode
  *
  * @throws cudf::logic_error if `gather_map_list` size is not same as `source_column` size.
- * @throws cudf::logic_error if gather_map contains null values.
+ * @throws std::invalid_argument if gather_map contains null values.
```

Comment:
Should the line below be updated with `dtype_error` (and the corresponding code)?

Reply:
No, because nullability is not a property of a data type; a column of any data type could contain nulls. The new exception is specifically designed for cases when an operation is attempted on a data type that doesn't make sense, e.g. trying to compute the sqrt of a string column.